### PR TITLE
WIP: Customize keyboard text

### PR DESCRIPTION
**Pull request by drbalar**
Via the backend, the keyboard/keypad text can now be customized. The font size does not seem changeable, just the text and its color.

image

### CODE REVIEW

File: app/public/static/js/components/keypad.js [hunk 1]
```diff
@@ -91,18 +92,25 @@ AFRAME.registerComponent("keypad", {
       this.solvedPuzzleEntity.setAttribute("visible", "true");
     } else {
       // Not solved yet
+      keypad_label = data.hasOwnProperty("keypad_text")
+        ? data.keypad_text
+        : "Enter Password";
       el.setAttribute("super-keyboard", {
-        label: "Enter Password",
+        label: keypad_label,
         labelColor: "black",
```
Comment (ahmedhamodi) [inline app/public/static/js/components/keypad.js:100]: Can we make this customizable? Or at the very least default to a non-black colour, it's a bit hard to see imo

Comment (drbalar) [reply app/public/static/js/components/keypad.js]: Made the default `white` and also customizable

**Comment by aaronabraham311** [comment]
Overall looks good! Is the text size not changeable throughout the entire application?

**Comment by drbalar** [comment]
> Overall looks good! Is the text size not changeable throughout the entire application?

No, i cant find `font-size` in the aframe docs. With the normal text we have to scale it up or down. For the keyboard's text, we have to scale the whole keyboard component since the text is a part of it.

File: app/public/static/js/components/keypad.js [hunk 2]
```diff
@@ -91,19 +92,32 @@ AFRAME.registerComponent("keypad", {
...
-        labelColor: "black",
+        label: keypad_text,
+        labelColor: keypad_text_color,
       });
     }
   },
 });
 
 async function removeError(el) {
+  const data = this.data.jsonData;
```
Comment (jaydhulia) [inline app/public/static/js/components/keypad.js:110]: Does `this` work here? Isn't `this` not defined? You may instead want to pass in the data in the method where we call `removeError`

Comment (drbalar) [reply app/public/static/js/components/keypad.js]: good catch

**Comment by jaydhulia** [comment]
`Is the text size not changeable throughout the entire application?`
@drbalar you can look at this: https://aframe.io/docs/1.1.0/components/text.html#sizing and https://aframe.io/docs/1.1.0/components/text.html#auto-scaling to look at text size and changing it. I recall changing it when I was playing around with it

**Comment by drbalar** [comment]
Thats a good find. Talked with Jay, Going to update a `title/header` text for all blackboard components. This way should be more consistent. Will create a new pr for it

**Pull request closed without merging**